### ISSUE TITLE
Can this model run on mobile devices?

### ISSUE THREAD

**Issue by ExplodedJelly** [ChaoningZhang/MobileSAM#32]
I'm considering the possibility of exporting MobileSam as an ONNX model and running it on mobile devices using onnxruntime.

**Comment by elblogbruno** [comment]
I am interested on this too!

**Comment by ivyas21** [comment]
check this out: https://github.com/ChaoningZhang/MobileSAM/pull/7

**Comment by ChaoningZhang** [comment]
> I'm considering the possibility of exporting MobileSam as an ONNX model and running it on mobile devices using onnxruntime.

We have implemented this. Please give it a check. If there are issues, please let us know.

**Comment by ChaoningZhang** [comment]
> 

We have implemented this and are now testing it before merging it. Stay tuned.

**Comment by ChaoningZhang** [comment]
> I am interested on this too!

We have implemented this and are now testing it before merging it. Stay tuned.

**Comment by ChaoningZhang** [comment]
> check this out: #7

Thanks for your help. We haven't got time for checking the pull request yet, however, we really appreciate your effort.

**Comment by fakeyee** [comment]
> > I am interested on this too!
> 
> We have implemented this and are now testing it before merging it. Stay tuned.

I'm wondering how is the progress about this great work?

**Comment by lizhiwen19900709** [comment]
@ChaoningZhang  exporting MobileSam as an ONNX model and running it on mobile devices using onnxruntime，Can you provide the example code?

**Comment by ryouchinsa** [comment]
Here is the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos/tree/master).
This code is originated from [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper) and implemented on macOS app [RectLabel](https://rectlabel.com). We customized the original code so that getMask() uses the previous mask result called as low_res_logits and retain the previous mask array for undo/redo actions. 

We hope this macOS version would help to develop the iOS version.
Please let us know your opinion.

![sam_polygon](https://github.com/ryouchinsa/sam-cpp-macos/assets/1954306/4640e139-c533-4b8c-b27b-e02a401b9bbd)

**Comment by ryouchinsa** [comment]
Improved "Create polygon using SAM" feature so that you can label pixels using the pixels option. You can label [Segment Anything 1 Billion (SA-1B)](https://ai.meta.com/datasets/segment-anything-downloads/) like dataset by yourself.

![sam-pixels](https://github.com/facebookresearch/segment-anything/assets/1954306/d828c18b-6fda-4258-b866-21bc34a0d17f)

**Comment by RobSchilderr** [comment]
> > I'm considering the possibility of exporting MobileSam as an ONNX model and running it on mobile devices using onnxruntime.
> 
> We have implemented this and are now testing it before merging it. Stay tuned.

Is this still coming?

**Comment by RobSchilderr** [comment]
> > I'm considering the possibility of exporting MobileSam as an ONNX model and running it on mobile devices using onnxruntime.
> 
> We have implemented this and are now testing it before merging it. Stay tuned.

Is this still coming?
